### PR TITLE
feat(surveys): Allow customers to edit zIndex of the survey popup

### PR DESCRIPTION
**Pull request by Phanatic**
## Problem

We've had the ability to [customize the zIndex of the survey popup](https://github.com/PostHog/posthog-js/blob/26150b197dc1a2f5241b8d29ee7a359331a1a0a4/src/extensions/surveys/surveys-utils.tsx#L29) in `posthog-js` for a while, but this setting wasn't exposed in the survey customization UI. 

Customer issue: https://posthoghelp.zendesk.com/agent/tickets/18485 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/21087)

## Changes

This PR adds a new input field for the zIndex of the survey popup.

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -2730,6 +2730,7 @@ export interface SurveyAppearance {
     thankYouMessageCloseButtonText?: string
     autoDisappear?: boolean
     position?: string
+    zIndex?: string
```
Comment (dmarticus) [inline frontend/src/types.ts:2733]: does this field already exist in the `SurveyAppearance` in posthog-js?

File: frontend/src/scenes/surveys/SurveyCustomization.tsx [hunk 2]
```diff
@@ -105,6 +105,13 @@ export function Customization({ appearance, surveyQuestionItem, onAppearanceChan
                     onChange={(submitButtonTextColor) => onAppearanceChange({ ...appearance, submitButtonTextColor })}
                     disabled={!surveysStylingAvailable}
                 />
+
+                <div className="mt-2">Survey form zIndex</div>
+                <LemonInput
+                    value={appearance?.zIndex}
+                    onChange={(zIndex) => onAppearanceChange({ ...appearance, zIndex })}
+                    disabled={!surveysStylingAvailable}
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/SurveyCustomization.tsx:113]: should we include a placeholder here?  The rest of the forms have one.

File: frontend/src/types.ts (hunk 1 above)
Comment (Phanatic) [reply frontend/src/types.ts:2733]: Yeah, here https://github.com/PostHog/posthog-js/blob/26150b197dc1a2f5241b8d29ee7a359331a1a0a4/src/extensions/surveys/surveys-utils.tsx#L29 

and here : 
https://github.com/PostHog/posthog-js/blob/26150b197dc1a2f5241b8d29ee7a359331a1a0a4/src/posthog-surveys-types.ts#L39

File: frontend/src/scenes/surveys/SurveyCustomization.tsx (hunk 2 above)
Comment (Phanatic) [reply frontend/src/scenes/surveys/SurveyCustomization.tsx:113]: good idea, done! 
image

Comment (dmarticus) [reply frontend/src/scenes/surveys/SurveyCustomization.tsx:113]: not to be nitpicky, but maybe an example value, too?  Or maybe a tooltip explaining what a zIndex is and the units in which its measured?  I know I'm being pedantic, I'm just not sure how much a PM e.g. would know what a zIndex is in this context.

File: frontend/src/types.ts (hunk 1 above)
Comment (dmarticus) [reply frontend/src/types.ts:2733]: huh, so weird we didn't have this capability already.

**Review by dmarticus — APPROVED** [review]

**Review by dmarticus — APPROVED** [review]

File: frontend/src/scenes/surveys/SurveyCustomization.tsx (hunk 2 above)
Comment (Phanatic) [reply frontend/src/scenes/surveys/SurveyCustomization.tsx:113]: The default is 99,999. I can add that here and put in a tooltip about what this is.

Comment (Phanatic) [reply frontend/src/scenes/surveys/SurveyCustomization.tsx:113]: image

**Pull request merged**